Gems/Atom/RPI/Code/Source/RPI.Reflect/Image/StreamingImageAsset.cpp

```diff
@@ -124,5 +213,45 @@ namespace AZ
 
             return mipChainAsset->GetSubImageData(mip - mipChain.m_mipOffset, slice);
         }
+
+        template<>
+        float StreamingImageAsset::GetSubImagePixelValue<float>(uint32_t mip, uint32_t slice, uint32_t x, uint32_t y, uint32_t componentIndex)
+        {
+            // TODO: Use the component index
+            (void)componentIndex;
+
+            auto imageData = GetSubImageData(mip, slice);
+
+            if (!imageData.empty())
+            {
+                const AZ::RHI::ImageDescriptor imageDescriptor = GetImageDescriptor();
+                auto width = imageDescriptor.m_size.m_width;
+                size_t index = (y * width) + x;
+
+                return Internal::RetrieveFloatValue(imageData.data(), index, imageDescriptor.m_format);
+            }
+
+            return 0.0f;
+        }
+
+        template<>
+        AZ::u32 StreamingImageAsset::GetSubImagePixelValue<AZ::u32>(uint32_t mip, uint32_t slice, uint32_t x, uint32_t y, uint32_t componentIndex)
+        {
+            // TODO: Use the component index
+            (void)componentIndex;
+
+            auto imageData = GetSubImageData(mip, slice);
+
+            if (!imageData.empty())
+            {
+                const AZ::RHI::ImageDescriptor imageDescriptor = GetImageDescriptor();
+                auto width = imageDescriptor.m_size.m_width;
+                size_t index = (y * width) + x;
```

Comment:
Could create a function for creating an index from (x, y, mip, slice) to dedupe with above code.